From a75db33cb47a588d291e2bb7b229b64662b21c6f Mon Sep 17 00:00:00 2001
From: Tad <tad@spotco.us>
Date: Fri, 5 Feb 2021 22:42:47 -0500
Subject: [PATCH] mvebu: venom resize kernel to 6MB

venom has a 3MB kernel partition as specified by the DTS.
3MB is not sufficient for building with many kernel modules or newer
kernel versions.

venom uboot however as set from factory will load up to 6MB.
This can be observed by looking a uboot log:
	NAND read: device 0 offset 0x900000, size 0x600000
	6291456 bytes read: OK
and from uboot environment variables:
	$ fw_printenv | grep "priKernSize";
	priKernSize=0x0600000

Resize the root partitions from 120MB to 117MB to let kernel expand
into it another 3MB.
And set kernel target size to 6MB.

Signed-off-by: Tad Davanzo <tad@spotco.us>
---
 .../files/arch/arm/boot/dts/armada-385-linksys-venom.dts  | 8 ++++----
 target/linux/mvebu/image/cortexa9.mk                      | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/target/linux/mvebu/files/arch/arm/boot/dts/armada-385-linksys-venom.dts b/target/linux/mvebu/files/arch/arm/boot/dts/armada-385-linksys-venom.dts
index de81600a80..a2ca3158cf 100644
--- a/target/linux/mvebu/files/arch/arm/boot/dts/armada-385-linksys-venom.dts
+++ b/target/linux/mvebu/files/arch/arm/boot/dts/armada-385-linksys-venom.dts
@@ -157,9 +157,9 @@
 			reg = <0x900000 0x7b00000>;	/* 123MB */
 		};
 
-		partition@c00000 {
+		partition@f00000 {
 			label = "rootfs1";
-			reg = <0xc00000 0x7800000>;	/* 120MB */
+			reg = <0xf00000 0x7500000>;	/* 117MB */
 		};
 
 		/* kernel2 overlaps with rootfs2 by design */
@@ -168,9 +168,9 @@
 			reg = <0x8400000 0x7b00000>;	/* 123MB */
 		};
 
-		partition@8700000 {
+		partition@8a00000 {
 			label = "rootfs2";
-			reg = <0x8700000 0x7800000>;	/* 120MB */
+			reg = <0x8a00000 0x7500000>;	/* 117MB */
 		};
 
 		/* last MB is for the BBT, not writable */
diff --git a/target/linux/mvebu/image/cortexa9.mk b/target/linux/mvebu/image/cortexa9.mk
index c07b064125..81728bf4a1 100644
--- a/target/linux/mvebu/image/cortexa9.mk
+++ b/target/linux/mvebu/image/cortexa9.mk
@@ -154,7 +154,7 @@ define Device/linksys_wrt32x
   DEVICE_ALT0_MODEL := Venom
   DEVICE_DTS := armada-385-linksys-venom
   DEVICE_PACKAGES += kmod-btmrvl kmod-mwifiex-sdio mwlwifi-firmware-88w8964
-  KERNEL_SIZE := 3072k
+  KERNEL_SIZE := 6144k
   KERNEL := kernel-bin | append-dtb
   SUPPORTED_DEVICES += armada-385-linksys-venom linksys,venom
   DEFAULT := n
-- 
2.29.2

